### PR TITLE
Added document parser for MS Office

### PR DESCRIPTION
**Pull request by oognuyh**
- Using Apache POI.
- Added test code and supported files.

### CODE REVIEW

File: langchain4j/src/main/java/dev/langchain4j/data/document/DocumentType.java [hunk 1]
```diff
@@ -4,5 +4,11 @@ public enum DocumentType {
 
     TXT,
     PDF,
-    HTML
+    HTML,
+    PPT,
+    PPTX,
```
Comment (langchain4j) [inline langchain4j/src/main/java/dev/langchain4j/data/document/DocumentType.java]: I am wondering, do we need to distinguish between PPT and PPTX (same for DOC/DOCX and XLS/XLSX) formats?
For html/htm/xhtml we only have one type: HTML.

File: langchain4j/src/main/java/dev/langchain4j/data/document/parser/MSOfficeDocumentParser.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+import static dev.langchain4j.internal.ValidationUtils.ensureNotNull;
+
+/**
+ * Extracts text from a Microsoft Office document.
+ * This parser supports various file formats, including ppt, pptx, doc, docx, xls, and xlsx.
+ * For detailed information on supported formats, please refer to the official Apache POI website:
+ * {@link <a href="https://poi.apache.org/">https://poi.apache.org/</a>}
+ */
+public class MSOfficeDocumentParser implements DocumentParser {
```
Comment (langchain4j) [inline langchain4j/src/main/java/dev/langchain4j/data/document/parser/MSOfficeDocumentParser.java]: Minor: could you please rename it to MsOffice... ?

**Comment by langchain4j** [comment]
@oognuyh thank you so much for your contribution! Great job!

File: langchain4j/src/main/java/dev/langchain4j/data/document/DocumentType.java (hunk 1 above)
Comment (oognuyh) [reply langchain4j/src/main/java/dev/langchain4j/data/document/DocumentType.java]: Oh! I was mistaken. I kept only the latest file extensions!

File: langchain4j/src/main/java/dev/langchain4j/data/document/parser/MSOfficeDocumentParser.java (hunk 2 above)
Comment (oognuyh) [reply langchain4j/src/main/java/dev/langchain4j/data/document/parser/MSOfficeDocumentParser.java]: Sure

File: langchain4j/src/main/java/dev/langchain4j/data/document/DocumentType.java (hunk 1 above)
Comment (langchain4j) [reply langchain4j/src/main/java/dev/langchain4j/data/document/DocumentType.java]: @oognuyh I see that you have removed tests for .doc .ppt .xls files, but why? These file extensions should be supported.
Sorry for miscommunication, what I meant is:
lets have only 3 types of documents in enum: DOC (for .doc and .docx), PPT (for .ppt and .pptx) and XLS (for .xls and .xlsx). Thanks!

Comment (oognuyh) [reply langchain4j/src/main/java/dev/langchain4j/data/document/DocumentType.java]: Ah, then I'll keep the existing test cases.

**Comment by langchain4j** [comment]
@oognuyh Thanks a lot! This PR is ready to mearge, the only thing is: there is now both MSOffice... and MsOffice... files for implementation and tests. Could you please remove MSOffice... ones? Thanks!

**Review by ghost — APPROVED** [review]
@oognuyh well done! Thank you for your contribution!

**Pull request merged**